src/api.ts

```diff
@@ -117,6 +117,7 @@ function makeFetcher(baseURL: string | URL, baseOptions: BaseOptions = {}) {
       ...ri,
       headers: mergeHeaders(
         typeof headers === 'function' ? await headers() : headers ?? {},
+        ri.headers ?? {},
         requestInit?.headers ?? {},
```

Comment:
Maybe we should decide on a better order.

1. Create make-service headers
2. Request Transform headers
3. User request headers

or

1. Create make-service headers
2. User request headers
3. Request Transform headers

Reply:
Hmm, I think @danielweinmann suggested it the other day and I can't remember why we didn't add it, do you remember, Dani?

Reply:
Hehe, can't remember 😅

Reply:
@iamandrewluca just to clarify, why do you need to set headers in the requestTransformer, can't it be done in the headers argument?
```ts
const service = makeService(BASE_URL, { headers: async () => { ... } })
```

I think the reason why we didn't allow it the other day is bc we couldn't figure out the better order and virtually anything could be possible through the `headers` function, I'd love to know if you found a blocker.

Reply:
I want to change the headers based on the request info. For example, to handle that FormData header

```ts
const requestTransformer: RequestTransformer = (request) => {
	const headers = new Headers(request.headers);

	if (!(request.body instanceof FormData)) {
		headers.set('Content-Type', 'application/json');
	}

	return { ...request, headers };
};

const headers = {
	Authorization: `Bearer ${session.getIdToken().getJwtToken()}`,
};

const service = makeService(import.meta.env.VITE_API_ORIGIN, {
	requestTransformer,
	headers,
});
```

Reply:
In that case I think it should have precedence over the `baseOptions` headers as it has more information to work with and can potentially be used for more specific decisions, WDYT?

If you agree and can make the change I'm all good with this PR ;)

Reply:
I think the current order (in PR) is ok 🤔 as the user still has control to override any headers if needed.

1. Create make-service headers
2. Request Transform headers
3. User request headers

Reply:
Exactly!